Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialSystemComponent.cpp

```diff
@@ -106,7 +106,13 @@ namespace AZ
             AzToolsFramework::EditorMenuNotificationBus::Handler::BusConnect();
             AzToolsFramework::EditorEvents::Bus::Handler::BusConnect();
             AzToolsFramework::ToolsApplicationNotificationBus::Handler::BusConnect();
-            AZ::SystemTickBus::Handler::BusConnect();
+            AzFramework::AssetCatalogEventBus::Handler::BusConnect();
+
+            // All material previews use the same model and lighting preset assets
+            // models/sphere.azmodel
+            m_materialPreviewModelAsset.Create(AZ::Data::AssetId("{6DE0E9A8-A1C7-5D0F-9407-4E627C1F223C}", 284780167));
+            // lightingpresets/thumbnail.lightingpreset.azasset
+            m_materialPreviewLightingPresetAsset.Create(AZ::Data::AssetId("{4F3761EF-E279-5FDD-98C3-EF90F924FBAC}", 0));
```

Comment:
Suggestion: Could the asset IDs get looked up using code like this once the OnCatalogLoaded is fired? It seems like it would be a bit more maintainable to have the asset names hardcoded instead of the asset IDs...

```
                    AZStd::string watchFolder;
                    AZ::Data::AssetInfo assetInfo;
                    AssetSystemRequestBus::BroadcastResult(
                        foundIt, &AssetSystemRequestBus::Events::GetSourceInfoBySourcePath, "models/sphere.azmodel", assetInfo,
                        watchFolder);

```

Reply:
I agree. I prefer readability and the hint to find the original file. Using the path should also continue to work if the asset ID format changes. However, there seems to be a push to remove hard coded asset paths. I thought I would get ahead of it.

It is worth noting that I had to go to the asset processor, look up the asset IDs, and paste them into code. That's not a very convenient workflow. In addition to that, I was originally using the function AZ::Data::AssetId::CreateString and pasted the value directly from the AP into the function. I discovered that it is parsing the asset ID sub ID as an unsigned long and assigning it to an unsigned int, which breaks for values out of the supported range. It was also explained that different systems display and serialize the sub ID in either base 10 or base 16. So the asset team is looking into that.